virttest/qemu_monitor.py

```diff
@@ -2565,9 +2566,10 @@ def balloon(self, size):
         # Send a balloon monitor command
         self.send_args_cmd("%s value=%s" % (cmd, size))
         # Look for BALLOON QMP events
-        if not utils_misc.wait_for(lambda: self.get_event(qmp_event), 120):
-            raise QMPEventError(cmd, qmp_event, self.vm.name, self.name)
-        logging.info("%s QMP event received" % qmp_event)
+        if check_event:
+            if not utils_misc.wait_for(lambda: self.get_event(qmp_event), 120):
+                raise QMPEventError(cmd, qmp_event, self.vm.name, self.name)
+            logging.info("%s QMP event received" % qmp_event)
```

Comment:
I'd suggest moving the event check from `Monitor`'s into `VM`'s

Reply:
The event can only be checked from qmp monitor other than hmp, I guess it's better to put it in class QMPMonitor().

Reply:
I add one condition check in it to make sure check event when monitor is qmp